src/manifold/src/boolean_result.cpp

```diff
@@ -499,16 +501,15 @@ void CreateProperties(Manifold::Impl &outR, const VecDH<TriRef> &refPQ,
 
     for (const int i : {0, 1, 2}) {
       const int vert = outR.halfedge_[3 * tri + i].startVert;
-      const glm::vec3 uvw = bary[3 * tri + i];
+      const glm::vec3 &uvw = bary[3 * tri + i];
 
-      auto key = std::make_tuple(PQ, -1, -1, -1);
+      glm::ivec4 key(PQ, vert, -1, -1);
       if (oldNumProp > 0) {
-        std::get<1>(key) = vert;
```

Comment:
These lines are something I just changed in #480 that I think we want to keep. The fact the tests still pass implies I need to add a test. This is to keep extra properties from being created when a mesh without properties is attached a mesh with properties.

Reply:
Ok but maybe I'll need to revert to hashmap for `manifold.cpp` as it breaks most assumptions.
I'll check.

Reply:
One way would be to use plain array for small size hashmap and hashmap for larger size.

Reply:
It's fine, I can still use the `std::vector<std::vector` approach.
I made the previous assumption that one prop could belong to only 1 logical vertex but it's not the case (although gets flattened in the end).